guardian/newsletters-nx
# Relaunch UX improvement

## What does this change?

Updates the Rendering for to give the Option to users to send a Braze update request where category is changed fro `article-based-legacy` -> article-based`.

Where the save succeeds, and the user has opted to send the notification, An email is sent and, where that send suceeds, the status of `brazeCampaignCreationStatus` is set to `REQUESTED`.


## How to test

Check out, set something to `article-based-legacy` and save. Now set to `article-based` .  You should see a modal after clicking update.  Choosing to save should trigger an email & set the brazeCampaignCreationStatus`  to `REQUESTED`. Choosing Cancel should do nothing. Chosing `Save without update` should just persist the changes in the form without triggering an email 

## How can we measure success?

By the above contions being true

## Have we considered potential risks?

Should be OK - It will generate emails to folks so we should be aware of this and:

1. Configure receipients sympathetically in CODE or..
2. Disable Email serives in CODE


## Images

**Happy Path....**

![success](https://github.com/guardian/newsletters-nx/assets/3277259/51bd9379-9a9f-4c1e-ab64-42de800f5c3e)

**Email Failure path....(mocked with expired token)**

![75sFailToBraze](https://github.com/guardian/newsletters-nx/assets/3277259/2b12aee1-6e54-453b-a664-8a4f08220c05)
